pkg/repository/provider/unified_repo.go

```diff
@@ -105,22 +99,104 @@ func (urp *unifiedRepoProvider) InitRepo(ctx context.Context, param RepoParam) e
 }
 
 func (urp *unifiedRepoProvider) ConnectToRepo(ctx context.Context, param RepoParam) error {
-	///TODO
+	log := urp.log.WithFields(logrus.Fields{
+		"BSL name": param.BackupLocation.Name,
+		"BSL UID":  param.BackupLocation.UID,
+	})
+
+	log.Debug("Start to connect repo")
+
+	repoOption, err := urp.getRepoOption(param)
+	if err != nil {
+		return errors.Wrap(err, "error to get repo options")
+	}
+
+	err = urp.repoService.Init(ctx, repoOption, false)
+	if err != nil {
+		return errors.Wrap(err, "error to connect backup repo")
+	}
+
+	log.Debug("Connect repo complete")
+
 	return nil
 }
 
 func (urp *unifiedRepoProvider) PrepareRepo(ctx context.Context, param RepoParam) error {
-	///TODO
+	log := urp.log.WithFields(logrus.Fields{
+		"BSL name": param.BackupLocation.Name,
+		"BSL UID":  param.BackupLocation.UID,
+	})
+
+	log.Debug("Start to prepare repo")
+
+	repoOption, err := urp.getRepoOption(param)
+	if err != nil {
+		return errors.Wrap(err, "error to get repo options")
+	}
+
+	err = urp.repoService.Init(ctx, repoOption, false)
+	if err == nil {
+		log.Debug("Repo has already been initialized remotely")
+		return nil
+	}
+
+	err = urp.repoService.Init(ctx, repoOption, true)
+	if err != nil {
+		return errors.Wrap(err, "error to init backup repo")
+	}
+
+	log.Debug("Prepare repo complete")
+
 	return nil
 }
 
 func (urp *unifiedRepoProvider) PruneRepo(ctx context.Context, param RepoParam) error {
-	///TODO
+	log := urp.log.WithFields(logrus.Fields{
+		"BSL name": param.BackupLocation.Name,
+		"BSL UID":  param.BackupLocation.UID,
+	})
+
+	log.Debug("Start to prune repo")
+
+	repoOption, err := urp.getRepoOptionForOpen(param, "full maintenance")
+	if err != nil {
+		return errors.Wrap(err, "error to get repo options")
+	}
+
+	repoOption.GeneralOptions[udmrepo.GenOptionMaintainMode] = udmrepo.GenOptionMaintainFull
+
+	err = urp.repoService.Maintain(ctx, repoOption)
+	if err != nil {
+		return errors.Wrap(err, "error to prune backup repo")
+	}
+
+	log.Debug("Prune repo complete")
+
 	return nil
 }
 
 func (urp *unifiedRepoProvider) PruneRepoQuick(ctx context.Context, param RepoParam) error {
-	///TODO
+	log := urp.log.WithFields(logrus.Fields{
+		"BSL name": param.BackupLocation.Name,
+		"BSL UID":  param.BackupLocation.UID,
+	})
+
+	log.Debug("Start to prune repo quick")
+
+	repoOption, err := urp.getRepoOptionForOpen(param, "quick maintenance")
```

Comment:
See my comment in the func `getRepoOptionForOpen` I think we should also set the value of `purpose` as a group of constants?

Reply:
Done